Map/LightOverlay.gd

```diff
@@ -0,0 +1,77 @@
+extends Node2D
+class_name LightOverlay
+
+var color_array: PackedColorArray
+
+var room_container: ColorRect
+var room_ui_array: Array[Array]
+
+var room_polygons_to_draw: Array[Rect2]
+
+func _init(_room_container: ColorRect, _room_ui_array: Array[Array]):
+	room_container = _room_container
+	room_ui_array = _room_ui_array
+
+func _draw():
+	var floor_rect_packed_array: PackedVector2Array
+	var room_circles: Array[PackedVector2Array]
+	var lit_rooms_to_draw: Array[RoomUI]
+	
+	# Account for minimum map placement T____T
+	for floor_index: int in range(room_ui_array.size()):
+		var floor_array: Array = room_ui_array[floor_index]
+		var _first_room_in_array: bool = false
+		
+		for room_index: int in range(floor_array.size()):
```

Comment:
same question as previously, why not use a for loop over the elements directly ?

Reply:
Ah good catch, it was some leftover code when I was trying to calculate each individual room position... Although in the latest push, I had to leave floor_index in order to help with some calculations on the darkness overlay portion.

Reply:
I only see `floor_index` being used on a check that does
```GDScript
if floor_index == 0:
	first_room_rect = room.get_room_rect()
```
I think you could do a for loop directly on the array without the `floor_index` and setting `first_room_rect` to a default value before the loop over a given floor, then checking against this default value in the loop. So if the value is still the default, you know it's the first room of the floor